scripts/firedrake-install

```diff
@@ -788,7 +806,13 @@ def pip_requirements(package):
             # loopy dependencies are installed in setup.py
             return
         else:
-            run_pip_install(["-r", "%s/requirements.txt" % package])
+            reqs = get_requirements("%s/requirements.txt" % package)
+            for req in reqs:
+                if re.split('<|>|=', req)[0] in parallel_packages:
+                    with environment(**parallel_compiler_env):
+                        run_pip_install([req])
+                else:
+                    run_pip_install([req])
```

Comment:
Please refactor so as to avoid the repeated code.

Reply:
Thank you. Refactored.